cmd/pow/cmd/deals_retrievals.go

```diff
@@ -0,0 +1,70 @@
+package cmd
+
+import (
+	"context"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/caarlos0/spin"
+	"github.com/logrusorgru/aurora"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/textileio/powergate/api/client"
+)
+
+func init() {
+	retrievalsCmd.Flags().BoolP("ascending", "a", false, "sort records ascending, default is descending")
+	retrievalsCmd.Flags().StringSlice("cids", []string{}, "limit the records to deals for the specified data cids")
+	retrievalsCmd.Flags().StringSlice("addrs", []string{}, "limit the records to deals initiated from  the specified wallet addresses")
+
+	dealsCmd.AddCommand(retrievalsCmd)
+}
+
+var retrievalsCmd = &cobra.Command{
+	Use:   "retrievals",
+	Short: "List retrieval records",
+	Long:  `List retrieval records`,
+	PreRun: func(cmd *cobra.Command, args []string) {
+		err := viper.BindPFlags(cmd.Flags())
+		checkErr(err)
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx, cancel := context.WithTimeout(context.Background(), cmdTimeout)
+		defer cancel()
+
+		var opts []client.ListDealRecordsOption
+
+		if viper.IsSet("ascending") {
```

Comment:
Nice viper method to check if a flag was actually set. Otherwise, we don't specify the option and go with whatever default exists on the back end